src/network/networkmanager.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <functional>
+
+#include <QLoggingCategory>
+
+class QJSEngine;
+class QQmlEngine;
+
+Q_DECLARE_LOGGING_CATEGORY(NETWORKMANAGER)
+
+/**
+ * @brief Manage the project NetworkManager
+ *
+ */
+class NetworkManager : public QObject
+{
+    Q_OBJECT
+public:
+    /**
+     * @brief Return NetworkManager pointer
+     *
+     * @return NetworkManager*
+     */
+    static NetworkManager* self();
+
+    /**
+     * @brief Do json requests
+     * @param url URL to be requested
+     * @param function that will be called
+     *
+     */
+    static void requestJson(const QUrl& url, std::function<void(QJsonDocument&)> function);
```

Comment:
`function` -> `callback`?

Reply:
Yes, that's the idea. I'm using the function name because this variable is a std::function. A `callback`variable can be many things and a function is a function.